### PR TITLE
Adding Session-related methods and types to

### PR DESCRIPTION
**Pull request by KKhurin**
- System.ServiceModel.Primitives
- System.Private.ServiceModel

### CODE REVIEW

File: src/System.ServiceModel.Primitives/tests/Description/OperationContractAttributeTest.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public static void IsInitiating_Method_Requires_Proper_SessionMode()
+    {
+        var channelFactory = CreateChannelFactoryHelper<INonInitiatingNonTerminatingService>("net.tcp://localhost/dummy.svc");
+        // The default ServiceContract's SessionMode.Allowed is not compatible with an OperationContract that has IsInitiating or IsTerminating
+        var exception = Assert.Throws<InvalidOperationException>(() =>
+        {
+            channelFactory.CreateChannel();
+        });
+        Assert.True(exception.Message.Contains("The contract 'INonInitiatingNonTerminatingService' is not self-consistent"));
+    }
```
Comment (mconnew) [inline src/System.ServiceModel.Primitives/tests/Description/OperationContractAttributeTest.cs]: This test should fail on systems running in a different locale if/when localization is working. Just check for the existence of the interface name in the message.

Comment (KKhurin) [inline src/System.ServiceModel.Primitives/tests/Description/OperationContractAttributeTest.cs]: Good point. Thanks!

**Comment by mconnew** [comment]
Other than the minor nit, everything looks good.

**Comment by zhenlan** [comment]
@StephenBonikowsky can you see if this is causing any VSTS test failures?

**Comment by StephenBonikowsky** [comment]
Yeah, I think it killed all the System.ServiceModel.Primitives tests by causing a timeout.
I'll get a PR out to update the version of ServiceModel packages we use.
That should fix it.

**Comment by StephenBonikowsky** [comment]
PR #1853

**Comment by zhenlan** [comment]
Thanks @StephenBonikowsky.

**Pull request merged**